memory/hugepage_sanity.py

```diff
@@ -0,0 +1,58 @@
+#!/usr/bin/env python
+#
+# This program is free software; you can redistribute it and/or modify
+# it under the terms of the GNU General Public License as published by
+# the Free Software Foundation; either version 2 of the License, or
+# (at your option) any later version.
+#
+# This program is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.
+#
+# See LICENSE for more details.
+#
+# Copyright: 2018 IBM
+# Author: Harish <harish@linux.vnet.ibm.com>
+#
+
+
+import os
+import shutil
+from avocado import Test
+from avocado import main
+from avocado.utils import process, build
+from avocado.utils.software_manager import SoftwareManager
+
+
+class HugepageSanity(Test):
+    """
+    Test allocates given no-hugepages of given size and mmap's
+    using MAP_HUGETLB with corresponding hugepage sizes
+    """
+    def copyutil(self, file_name):
+        shutil.copyfile(os.path.join(self.datadir, file_name),
+                        os.path.join(self.teststmpdir, file_name))
+
+    def setUp(self):
+        smm = SoftwareManager()
+        self.hpagesize = int(self.params.get('hpagesize', default=''))
```

Comment:
can it has some default value ?

Reply:
A default value may be different from system's default hugepage. Hence, if it is empty the c script will use the default hugepage of the machine. It will be repetitive to do the same here.